MonoGame.Framework/Graphics/Texture2D.OpenGL.cs

```diff
@@ -70,73 +70,61 @@ public partial class Texture2D : Texture
         private void PlatformConstruct(int width, int height, bool mipmap, SurfaceFormat format, SurfaceType type, bool shared)
         {
             this.glTarget = TextureTarget.Texture2D;
+            format.GetGLFormat(GraphicsDevice, out glInternalFormat, out glFormat, out glType);
+            // GL texture is generated on first access in PlatformSetData
```

Comment:
Actually a more definitive test.  Is video memory allocated or not?

On DirectX after `new Texture2D(...)` is called video memory is allocated.

I suspect in this case no video memory has been allocated at all.

Looking at the GL docs on `GL.TexImage2D` it says:

https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glTexImage2D.xhtml
> data may be a null pointer. In this case, texture memory is allocated to accommodate a texture of width width and height height.

This suggests to me that not calling `GL.TexImage2D` means no texture memory has been allocated at all.  I've seen some suggestions that `GL.TexImage2D` can error out with `GL_OUT_OF_MEMORY`, but others suggest that it won't or at least doesn't happen on some platforms.

Also I would think that the texture memory needed for a texture would be allocated as one whole allocation for all mip levels.  Calling `GL.TexImage2D` once for each mip level would be a weird way to make a single allocation like that.  This suggests `GL.TexImage2D` doesn't allocate any memory and instead it is allocated sometime later when it has all the information needed.


Gosh GL is such a mess.